snow/consensus/snowman/bootstrapper/sampler.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bootstrapper
+
+import (
+	"github.com/ava-labs/avalanchego/utils/math"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+// Sample keys from [elements] uniformly by weight without replacement. The
+// returned set will have size less than or equal to [maxSize]. This function
+// will error if the sum of all weights overflows.
+func Sample[T comparable](elements map[T]uint64, maxSize int) (set.Set[T], error) {
```

Comment:
There wasn't really a need to make this generic... But I felt like it was just as clean as hardcoding `ids.NodeID`... 🤷

Reply:
If we wanted to, we could also make `Majority` and `Minority` generic. Same as here, though, it wouldn't really buy us anything.

Reply:
We could also think about either moving this into `utils` or making this package-private so discourage people from using `bootstrapper.Sample` to get a random subset of a map

Reply:
I was going to put this into the `sampler` package - but it caused a circular dependency because of the `SampleableSet`.

Reply:
Also it can't actually be package private - it is used in a different package